### ISSUE TITLE
MixedClusterClientYamlTestSuiteIT "Test cat indices with invalid health parameter" failing

### ISSUE THREAD

**Issue by przemekwitek** [elastic/elasticsearch#57119]
**Build scan**:

Build | 20200525054934-C910E98C
Log | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=fedora-29&&immutable/775/console

Build Scans
[7.7.1] https://gradle-enterprise.elastic.co/s/h5kjsvvdjosue
[7.9.0] https://gradle-enterprise.elastic.co/s/cxgh45dkzkqjo
[7.8.0] https://gradle-enterprise.elastic.co/s/bouk5updx6j26
[7.7.1] https://gradle-enterprise.elastic.co/s/rb3kfwbsan37y
[7.8.0] https://gradle-enterprise.elastic.co/s/7jw3by7bhpkei
[7.9.0] https://gradle-enterprise.elastic.co/s/nbl55ifh324so
https://gradle-enterprise.elastic.co/s/vbqd3m7kkghsq

**Repro line**:

```
REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v7.9.0#mixedClusterTest' --tests "org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" \
  -Dtests.method="test {p0=cat.indices/10_basic/Test cat indices with invalid health parameter}" \
  -Dtests.seed=6207F8AE9DD97C04 \
  -Dtests.security.manager=true \
  -Dtests.locale=fr-BE \
  -Dtests.timezone=Australia/Currie \
  -Dcompiler.java=14 \
  -Druntime.java=11

REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v7.9.0#mixedClusterTest' --tests "org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" \
  -Dtests.method="test {p0=cat.indices/10_basic/Test cat indices with invalid health parameter}" \
  -Dtests.seed=6207F8AE9DD97C04 \
  -Dtests.security.manager=true \
  -Dtests.locale=fr-BE \
  -Dtests.timezone=Australia/Currie \
  -Dcompiler.java=14 \
  -Druntime.java=11
```

**Reproduces locally?**:

No

**Applicable branches**:

master

**Failure history**:

https://build-stats.elastic.co/app/kibana#/dashboard/02b9d310-9513-11e8-a9c0-db5f285c257f?_g=(refreshInterval:(pause:!f,value:10000),time:(from:now-14d%2Fd,mode:relative,to:now-0d%2Fd))&_a=(description:'',filters:!(),fullScreenMode:!f,options:(darkTheme:!f,hidePanelTitles:!f,useMargins:!t),panels:!((embeddableConfig:(vis:(legendOpen:!t)),gridData:(h:15,i:'1',w:48,x:0,y:0),id:d471ae10-9512-11e8-a9c0-db5f285c257f,panelIndex:'1',type:visualization,version:'6.3.1'),(embeddableConfig:(spy:!n,vis:(legendOpen:!f)),gridData:(h:13,i:'4',w:7,x:21,y:28),id:'38f6aff0-977c-11e8-a9c0-db5f285c257f',panelIndex:'4',type:visualization,version:'6.3.1'),(embeddableConfig:(vis:(colors:('process.status:FAILURE':%23890F02,'process.status:SUCCESS':%23629E51),legendOpen:!f)),gridData:(h:13,i:'6',w:24,x:0,y:15),id:'65d72d70-f1b7-11e8-ba96-eb5c43ee553c',panelIndex:'6',type:visualization,version:'6.3.2'),(embeddableConfig:(vis:(colors:(FAILURE:%23890F02,SUCCESS:%23508642,'process.status:FAILURE':%23890F02,'process.status:SUCCESS':%23629E51),legendOpen:!f)),gridData:(h:13,i:'7',w:21,x:0,y:28),id:'2373fab0-f27c-11e8-ba96-eb5c43ee553c',panelIndex:'7',type:visualization,version:'6.3.2'),(embeddableConfig:(),gridData:(h:13,i:'8',w:24,x:24,y:15),id:a9ae1c00-83e9-11ea-a0b8-098e8b2ee137,panelIndex:'8',type:visualization,version:'6.8.3'),(embeddableConfig:(),gridData:(h:22,i:'9',w:20,x:28,y:28),id:'966886c0-83ea-11ea-a0b8-098e8b2ee137',panelIndex:'9',type:visualization,version:'6.8.3'),(embeddableConfig:(),gridData:(h:50,i:'10',w:28,x:0,y:41),id:f64870f0-83ea-11ea-a0b8-098e8b2ee137,panelIndex:'10',type:search,version:'6.8.3')),query:(language:lucene,query:MixedClusterClientYamlTestSuiteIT),timeRestore:!t,title:'%5Bkbn%5D%20Failing%20Tests',viewMode:view)

**Failure excerpt**:

```
09:15:43 org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=cat.indices/10_basic/Test cat indices with invalid health parameter} FAILED
09:15:43     java.lang.RuntimeException: Failure at [cat.indices/10_basic:288]: 60.000 milliseconds timeout on connection http-outgoing-28 [ACTIVE]
09:15:43         at __randomizedtesting.SeedInfo.seed([6207F8AE9DD97C04:EA53C774332511FC]:0)
09:15:43         at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:407)
09:15:43         at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:382)
09:15:43         at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
09:15:43         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
09:15:43         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
09:15:43         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
09:15:43         at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
09:15:43         at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
09:15:43         at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
09:15:43         at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
09:15:43         at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
09:15:43         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
09:15:43         at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
09:15:43         at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
09:15:43         at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
09:15:43         at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
09:15:43         at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
09:15:43         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
09:15:43         at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
09:15:43         at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
09:15:43         at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
09:15:43         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
09:15:43         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
09:15:43         at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
09:15:43         at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
09:15:43         at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
09:15:43         at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
09:15:43         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
09:15:43         at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
09:15:43         at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
09:15:43         at java.base/java.lang.Thread.run(Thread.java:834)
09:15:43 
09:15:43         Caused by:
09:15:43         java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-28 [ACTIVE]
09:15:43             at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:818)
09:15:43             at org.elasticsearch.client.RestClient.performRequest(RestClient.java:250)
09:15:43             at org.elasticsearch.client.RestClient.performRequest(RestClient.java:237)
09:15:43             at org.elasticsearch.test.rest.yaml.ClientYamlTestClient.callApi(ClientYamlTestClient.java:186)
09:15:43             at org.elasticsearch.test.rest.yaml.ClientYamlTestExecutionContext.callApiInternal(ClientYamlTestExecutionContext.java:169)
09:15:43             at org.elasticsearch.test.rest.yaml.ClientYamlTestExecutionContext.callApi(ClientYamlTestExecutionContext.java:101)
09:15:43             at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:280)
09:15:43             at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:398)
09:15:43             ... 37 more
09:15:43 
09:15:43             Caused by:
09:15:43             java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-28 [ACTIVE]
09:15:43                 at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
09:15:43                 at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92)
09:15:43                 at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39)
09:15:43                 at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
09:15:43                 at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
09:15:43                 at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502)
09:15:43                 at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
09:15:43                 at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
09:15:43                 at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
09:15:43                 at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
09:15:43                 ... 1 more
```

**Comment by danhermann** [comment]
I believe this should be resolved by the backport here: https://github.com/elastic/elasticsearch/pull/57120

**Comment by markharwood** [comment]
@danhermann  I think I just hit this same failure in a 7.x PR build which includes #57120

**Comment by danhermann** [comment]
Thanks, @markharwood. I will look into that one as well.

**Comment by markharwood** [comment]
Thanks. My PR build failure is [here](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+default-distro/24461/) in case you need it.

**Comment by danhermann** [comment]
I backported this fix to 7.7.1 which should address the failures in the mixed cluster tests.

**Comment by gwbrown** [comment]
This failure is still occurring in 7.x, mostly in BWC tests. Example:
[Failure when testing BWC against 7.4.2](https://gradle-enterprise.elastic.co/s/67djqwev3swwy/tests/:qa:mixed-cluster:v7.4.2%23mixedClusterTest/org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT/test%20%7Bp0=cat.indices%2F10_basic%2FTest%20cat%20indices%20with%20invalid%20health%20parameter%7D).
